File: src/main/java/com/demigodsrpg/chitchat/tag/NameTag.java
package com.demigodsrpg.chitchat.tag;

import net.kyori.adventure.text.Component;
import org.bukkit.entity.Player;

public class NameTag extends PlayerTag {
    @Override
    public String getName() {
        return "name";
    }

    @Override
    public int getPriority() {
        return 999;
    }

    @Override
    public Component getComponentFor(Player tagSource) {
        return tagSource.displayName();
    }
}

File: src/main/kotlin/com/demigodsrpg/chitchat/tag/NameTag.kt
package com.demigodsrpg.chitchat.tag

import net.md_5.bungee.api.chat.TextComponent
import org.bukkit.entity.Player

class NameTag : PlayerTag() {
    override val name: String
        get() = "name"

    override val priority: Int
        get() = 999

    override fun getComponentFor(tagSource: Player): TextComponent? {
        val ret = TextComponent("")
        for (component in TextComponent.fromLegacyText(tagSource.displayName)) {
            ret.addExtra(component)
        }
        return ret
    }
}
